command/scaling_policy_list.go

```diff
@@ -0,0 +1,151 @@
+package command
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+// Ensure ScalingPolicyListCommand satisfies the cli.Command interface.
+var _ cli.Command = &ScalingPolicyListCommand{}
+
+// ScalingPolicyListCommand implements cli.Command.
+type ScalingPolicyListCommand struct {
+	Meta
+}
+
+// Help satisfies the cli.Command Help function.
+func (s *ScalingPolicyListCommand) Help() string {
+	helpText := `
+Usage: nomad scaling policy list [options]
+
+  List is used to list the currently configured scaling policies.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Policy Info Options:
+
+  -json
+    Output the scaling policy in its JSON format.
+
+  -t
+    Format and display the scaling policy using a Go template.
+`
+	return strings.TrimSpace(helpText)
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (s *ScalingPolicyListCommand) Synopsis() string {
+	return "Display all Nomad scaling policies"
+}
+
+func (s *ScalingPolicyListCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(s.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-json": complete.PredictNothing,
+			"-t":    complete.PredictAnything,
+		})
+}
+
+// Name returns the name of this command.
+func (s *ScalingPolicyListCommand) Name() string { return "scaling policy list" }
+
+// Run satisfies the cli.Command Run function.
+func (s *ScalingPolicyListCommand) Run(args []string) int {
+	var json bool
+	var tmpl string
+
+	flags := s.Meta.FlagSet(s.Name(), FlagSetClient)
+	flags.Usage = func() { s.Ui.Output(s.Help()) }
+	flags.BoolVar(&json, "json", false, "")
+	flags.StringVar(&tmpl, "t", "", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if args = flags.Args(); len(args) > 0 {
+		s.Ui.Error("This command takes no arguments")
+		s.Ui.Error(commandErrorText(s))
+	}
+
+	// Get the HTTP client.
+	client, err := s.Meta.Client()
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	policies, _, err := client.Scaling().ListPolicies(nil)
```

Comment:
I think we need a `-namespace` argument for listing policies?

Reply:
Namespace is included in the command meta when using `FlagSetClient`. The client subsequently gets its config correctly populated [here](https://github.com/hashicorp/nomad/blob/97d03887c767c0e9c78a2cdd15825544c2010dbe/command/meta.go#L124).

Reply:
Magical!

```shellsession
$ nomad scaling policy list -namespace=default
ID                                    Enabled  Target
52cdb238-c37c-662a-5c6e-7af1b5978757  true     Namespace:default,,
aaf6a549-80c4-8716-f70f-b59d5cd0395d  false    Namespace:default,Job:webapp,Group:demo

$ nomad scaling policy list -namespace=dev
ID                                    Enabled  Target
2b31db60-ca35-e992-d223-527e82b850aa  true     Namespace:dev,,
````

Reply:
Note: trailing commas are now removed.